test/unit/connection_string.test.ts

```diff
@@ -89,6 +89,15 @@ describe('Connection String', function () {
     expect(options.credentials.source).to.equal('$external');
   });
 
+  it('should retain user-specified authSource with no credentials for non-srv URI', function () {
+    const mockAuthSource = 'thisShouldBeAuthSource';
+    const options = parseOptions(`mongodb://localhost/?authSource=${mockAuthSource}`);
```

Comment:
Just commenting so we don't lose track, we're going to want to add an integration test for this, because the error that gets thrown is all the way down at the connection creation layer.